code/datums/ai/kitchenbot/kitchenbot_behaviors.dm

```diff
@@ -0,0 +1,166 @@
+	var/BB_sound = controller.blackboard[BB_KITCHENBOT_TASK_SOUND]
+	controller.pawn.audible_message("<span class='hear'>[controller.pawn] [BB_text]!</span>")
+	playsound(controller.pawn, BB_sound, 50, FALSE)
+
+//GRIDDLING
+
+/datum/ai_behavior/find_and_set/find_griddle
+	locate_path = /obj/machinery/griddle
+	bb_key_to_set = BB_KITCHENBOT_CHOSEN_GRIDDLE
+
+/datum/ai_behavior/find_and_set/find_griddle/finish_action(datum/ai_controller/controller, succeeded)
+	. = ..()
+	if(succeeded)
+		var/BB_text = controller.blackboard[BB_KITCHENBOT_TASK_TEXT]
+		var/BB_sound = controller.blackboard[BB_KITCHENBOT_TASK_SOUND]
+		controller.pawn.audible_message("<span class='hear'>[controller.pawn] [BB_text]! It must have located a griddle.</span>")
+		playsound(controller.pawn, BB_sound, 50, FALSE)
+
+/datum/ai_behavior/find_and_set/find_stockpile
+	locate_path = /obj/structure/holosign/kitchenbot_stockpile
+	bb_key_to_set = BB_KITCHENBOT_CHOSEN_STOCKPILE
+
+/datum/ai_behavior/find_and_set/find_stockpile/finish_action(datum/ai_controller/controller, succeeded)
+	. = ..()
+	if(succeeded)
+		var/BB_text = controller.blackboard[BB_KITCHENBOT_TASK_TEXT]
+		var/BB_sound = controller.blackboard[BB_KITCHENBOT_TASK_SOUND]
+		controller.pawn.audible_message("<span class='hear'>[controller.pawn] [BB_text]! It must have located a stockpile.</span>")
+		playsound(controller.pawn, BB_sound, 50, FALSE)
+
+//kitchenbot exclusive
+/datum/ai_behavior/find_and_set/find_stockpile_target
+	bb_key_to_set = BB_KITCHENBOT_TARGET_IN_STOCKPILE
+
+/datum/ai_behavior/find_and_set/find_stockpile_target/search_tactic(datum/ai_controller/controller)
+	var/datum/ai_controller/kitchenbot/kitchenbot_controller = controller
+	var/obj/stockpile = kitchenbot_controller.blackboard[BB_KITCHENBOT_CHOSEN_STOCKPILE]
+	var/turf/stockpile_turf = get_turf(stockpile)
+	if(!stockpile_turf)
+		return
+	var/list/should_griddle = list()
+	for(var/obj/item/grillable in stockpile_turf.contents)
+		var/list/banned_items = kitchenbot_controller.blackboard[BB_KITCHENBOT_ITEMS_BANNED]
+		if(grillable in banned_items)
+			continue
+		var/datum/component/grillable/grill_comp = grillable.GetComponent(/datum/component/grillable) //change this for final merge
+		if(!grill_comp || !grill_comp.positive_result)//bad, don't grill this
+			banned_items += grillable
+			continue
+		should_griddle += grillable
+	return pick_n_take(should_griddle)
+
+/datum/ai_behavior/forcemove_grab/grab_griddlable
+	bb_key_target = BB_KITCHENBOT_TARGET_IN_STOCKPILE
+
+
+/datum/ai_behavior/put_on_grill
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT
+	required_distance = 1
+
+/datum/ai_behavior/put_on_grill/perform(delta_time, datum/ai_controller/controller)
+	. = ..()
+
+	var/obj/machinery/griddle/griddle = controller.blackboard[BB_KITCHENBOT_CHOSEN_GRIDDLE]
+	var/obj/item/grillable = controller.blackboard[BB_KITCHENBOT_TARGET_IN_STOCKPILE]
+
+	grillable.forceMove(griddle)
+	griddle.AddToGrill(grillable, controller.pawn)
+	if(!griddle.on)
```

Comment:
make this a proc on the griddle you vile animal, make it the same one a normal human uses.